basics/indexing/dictionary-index.md

```diff
@@ -0,0 +1,124 @@
+| noDictionarySizeRatioThreshold | 0.85    | The ratio used in the heuristics.                                    |
+
+It is important to note that these parameters are configured for all columns on the table.
+When this optimization is enabled, columns explicitly marked as dictionary will actually be encoded as raw 
+if:
+- They are not multi-valued.
+- And their column type is fixed size (like int, long, double, timestamp, etc).
+- And their ratio between the forward index size encoded as raw and the forward index size encoded as dictionary is lower than `noDictionarySizeRatioThreshold`.
+
+If `optimizeDictionary` is false and `optimizeDictionaryForMetrics` is true, only columns declared as metric will be affected.
+If `optimizeDictionary` is true, all columns will be affected.
+Also in this case columns indexed as [text](./text-search-support.md) or [JSON](./json-index.md) will automatically be indexed as raw.
+Note that `optimizeDictionary` has priority over `optimizeDictionaryForMetrics` and the special rule applied to text 
+and JSON indexes only applies when `optimizeDictionary` is true.
+
+### Parameters
+
+There are some options you can configure:
+
+| Parameter              | Default | Description                                    |
+|------------------------|---------|------------------------------------------------|
+| onHeap                 | false   | If the index must be load on heap or off heap. |
+| useVarLengthDictionary | false   | How to store variable length values.           |
+
+Dictionaries are always off heap, but in cases where the cardinality is small and the footprint on heap is acceptably 
+small, they can be copied in memory by changing `onHeap` parameter to true.
+When they are on heap, dictionaries can be faster and some extra optimizations can be done.
+
+Parameter `useVarLengthDictionary` only affects columns whose values require variable number of bytes.
+That means that:
+* The column type requires variable number of bytes (like string, bytes or big decimal).
+* Not all values on the segment require the same number of bytes.
+  For example, if the segment only contains values "a", "b" and "c", Pinot will detect that there all values
+  can be represented with the same number of bytes.
+
+By default, this parameter is `false`, which means that Pinot will calculate the larger value contained in the segment.
+That is the length that will be used for each value.
+This guarantees that all values can be stored and produce faster access and a more compressed layout when the length of
+values are quite similar.
+
+This default is not great when there are a few very large values and a lot of very small ones.
+In that case it is recommended to indicate Pinot to use a variable length encoding by setting `useVarLengthDictionary` to `true`.
+When the variable encoding is used, Pinot needs to store the entry length, so the cost of storing an entry is
+their actual size plus 4 bytes offset.
```

Comment:
the actual size of the entry?

Reply:
It should be `its` given it we are talking about a singular entity (an entry), right?